rust/src/sip/sip.rs

```diff
@@ -147,20 +148,76 @@ impl SIPState {
         }
     }
 
+    fn parse_request_tcp(
+        &mut self, flow: *const core::Flow, stream_slice: StreamSlice,
+    ) -> AppLayerResult {
+        let input = stream_slice.as_slice();
+        if input.is_empty() {
+            return AppLayerResult::ok();
+        }
+
+        let mut start = input;
+        while !start.is_empty() {
+            match sip_parse_request(start) {
+                Ok((rem, request)) => {
+                    let pdu_len = (start.len() - rem.len()) as i64;
+                    let _f = Frame::new(
```

Comment:
the frame should be created earlier, and not depend on the parser returning ok. It should be created as soon as we start parsing the data, and then be updated with the length when it's complete

Reply:
Sorry for the confusion, I was the one who asked this.

What should the length be updated to when the parser errors (incomplete or not) ?

Why does it have to be so ?
On the S-V test, a frame was created before returning incomplete, and suricata did not match on it... 
So, why should we create a frame if we cannot match on it ? (or how can we match on it)

Reply:
Indeed, the telnet code seems to do it well.

We need this so we can start matching immediately for large frames. Also if we have malformed data we can still do frame based content matching.

Reply:
> We need this so we can start matching immediately for large frames. Also if we have malformed data we can still do frame based content matching.

This does not seem to work cf https://github.com/OISF/suricata/pull/10037 CI is green with SV https://github.com/OISF/suricata-verify/pull/1538 where there is only one alert for https://github.com/OISF/suricata-verify/pull/1538/files#diff-efc7fbd99eb9389e40b2f9f16aa9eca8d58a80ea88324756b8a5d4b476cf69cfR18 `alert sip any any -> any any (flow:to_server; frame:pdu; content:"REGISTER"; startswith; sid:2;)` after the frame is created on packet 6 (and no alert for the frame created on packet 4, as can be checked adding a `bsize` keyword to the rule)